src/sql/src/pure/sql_server.rs

```diff
@@ -414,26 +414,70 @@ pub(super) fn generate_source_export_statement_values(
     };
 
     // Filter out columns that the user wanted to exclude.
-    let included_columns = table.columns.iter().filter_map(|c| {
-        c.column_type
-            .as_ref()
-            .map(|ct| (c.name.as_ref(), ct, c.primary_key_constraint.clone()))
-    });
+    let included_columns = table
+        .columns
+        .iter()
+        .filter_map(|c| c.column_type.as_ref().map(|ct| (c.name.as_ref(), ct)));
 
-    let mut primary_keys: BTreeMap<Arc<str>, Vec<Ident>> = BTreeMap::new();
     let mut column_defs = vec![];
     let mut constraints = vec![];
 
-    for (col_name, col_type, col_primary_key_constraint) in included_columns {
+    // there are likely less excluded columns than included
+    let excluded_columns: BTreeSet<_> = table
+        .columns
+        .iter()
+        .filter_map(|c| match &c.column_type {
+            None => Some(c.name.as_ref()),
+            Some(_) => None,
+        })
+        .collect();
+    for constraint in table.constraints.iter() {
+        // if one or more columns of the constraint are excluded, do not record the constraint
+        // in materialize
+        if constraint
+            .column_names
+            .iter()
+            .any(|col| excluded_columns.contains(col.as_str()))
+        {
+            tracing::debug!(
+                "skipping constraint {name} due to excluded column",
+                name = &constraint.constraint_name
+            );
+            continue;
+        }
+
+        constraints.push(mz_sql_parser::ast::TableConstraint::Unique {
+            name: Some(Ident::new_lossy(constraint.constraint_name.clone())),
+            columns: constraint
+                .column_names
+                .iter()
+                .map(Ident::new)
+                .collect::<Result<_, _>>()?,
+            is_primary: matches!(
+                constraint.constraint_type,
+                SqlServerTableConstraintType::PrimaryKey
+            ),
+            // NULLs are distinct for SQL Server UNIQUE constraint, and not allowed for PRIMARY KEY.
```

Comment:
The comment should say that NULLs are *not* distinct, right? That's what I understand from the link to the docs.